src/bundle/ui-dev/src/modules/common/popup-menu/popup.menu.js

```diff
@@ -0,0 +1,194 @@
+                    <div className="ibexa-input-text-wrapper__actions">
+                        <button
+                            type="button"
+                            className="btn ibexa-input-text-wrapper__action-btn ibexa-input-text-wrapper__action-btn--clear"
+                            tabIndex="-1"
+                            onClick={resetInputValue}
+                        >
+                            <Icon name="discard" extraClasses="ibexa-icon--tiny-small" />
+                        </button>
+                        <button
+                            type="button"
+                            className="btn ibexa-input-text-wrapper__action-btn ibexa-input-text-wrapper__action-btn--search"
+                            tabIndex="-1"
+                        >
+                            <Icon name="search" extraClasses="ibexa-icon--small" />
+                        </button>
+                    </div>
+                </div>
+            </div>
+        );
+    };
+    const renderFooter = () => {
+        if (!footer) {
+            return null;
+        }
+
+        return <div className="c-popup-menu__footer">{footer}</div>;
+    };
+
+    useEffect(() => {
+        calculateAndSetItemsListStyles();
+        setIsRendered(true);
+
+        const onInteractionOutside = (event) => {
+            if (containerRef.current.contains(event.target) || referenceElement.contains(event.target)) {
+                return;
+            }
+
+            onClose();
+        };
+
+        window.document.body.addEventListener('click', onInteractionOutside, false);
+        scrollContainer.addEventListener('scroll', calculateAndSetItemsListStyles, false);
+
+        return () => {
+            window.document.body.removeEventListener('click', onInteractionOutside);
+            scrollContainer.removeEventListener('scroll', calculateAndSetItemsListStyles);
+
+            setItemsListStyles({});
+        };
+    }, [onClose, scrollContainer]);
+
+    return (
+        <div className={popupMenuClassName} style={itemsListStyles} ref={containerRef}>
+            {renderSearch()}
+            <div className="c-popup-menu__groups">{items.map(renderGroup)}</div>
+            {renderFooter()}
+        </div>
+    );
+};
+
+PopupMenu.propTypes = {
+    referenceElement: PropTypes.node.isRequired,
+    extraClasses: PropTypes.string,
+    footer: PropTypes.node,
+    items: PropTypes.arrayOf({
+        items: PropTypes.shape({
+            value: PropTypes.oneOf([PropTypes.string, PropTypes.number]),
+            label: PropTypes.string,
+        }),
+    }),
+    onClose: PropTypes.func,
+    onItemClick: PropTypes.func,
+    positionOffset: PropTypes.func,
+    scrollContainer: PropTypes.node,
+};
+
+PopupMenu.defaultProps = {
+    extraClasses: '',
+    footer: null,
+    items: [],
+    onClose: () => {},
+    onItemClick: () => {},
+    positionOffset: () => ({ x: 0, y: 0 }),
+    scrollContainer: window.document.body,
```

Comment:
Could you use helper function getRootDOMElement from https://github.com/ibexa/admin-ui/blob/main/src/bundle/Resources/public/js/scripts/helpers/context.helper.js#L62 ?